.github/workflows/ios-build-and-test.yml

```diff
@@ -0,0 +1,223 @@
+      run_test_ios_functional:
+        required: false
+        type: boolean
+        default: true
+      run_test_ios_integration:
+        required: false
+        type: boolean
+        default: true
+      run_test_tvos_unit:
+        required: false
+        type: boolean
+        default: true
+      run_test_tvos_functional:
+        required: false
+        type: boolean
+        default: true
+      run_build_xcframework_and_app:
+        required: false
+        type: boolean
+        default: true
+
+jobs:
+  validate-code:
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4.1.7
+
+      - name: Checkout actions from aepsdk-commons
+        uses: actions/checkout@v4.1.7
+        with:
+          repository: adobe/aepsdk-commons
+          ref: main
+          path: .github/aepsdk-commons
+          sparse-checkout: '.github/actions' # Only checkout the GitHub actions directory
+
+      - name: Setup Dependencies
+        uses: ./.github/aepsdk-commons/.github/actions/setup-action
+
+      - name: Remove files from aepsdk-commons
+        run: rm -rf ./github/aepsdk-commons
+
+      - name: Lint Source Code
+        run: make lint
+
+  test-ios-unit:
+    runs-on: macos-latest
+    needs: validate-code
+    if: inputs.run_test_ios_unit
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4.1.7
+
+      - name: Checkout actions from aepsdk-commons
+        uses: actions/checkout@v4.1.7
+        with:
+          repository: adobe/aepsdk-commons
+          ref: main
+          path: .github/aepsdk-commons
+          sparse-checkout: '.github/actions' # Only checkout the GitHub actions directory
+
+      - name: Setup Dependencies
+        uses: ./.github/aepsdk-commons/.github/actions/setup-action
+
+      - name: Remove files from aepsdk-commons
+        run: rm -rf ./github/aepsdk-commons
+
+      - name: Run iOS Unit Tests
+        run: make unit-test-ios
+
+  test-ios-functional:
+    runs-on: macos-latest
+    needs: validate-code
+    if: inputs.run_test_ios_functional
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4.1.7
+      
+      - name: Checkout actions from aepsdk-commons
+        uses: actions/checkout@v4.1.7
+        with:
+          repository: adobe/aepsdk-commons
+          ref: main
+          path: .github/aepsdk-commons
+          sparse-checkout: '.github/actions' # Only checkout the GitHub actions directory
+
+      - name: Setup Dependencies
+        uses: ./.github/aepsdk-commons/.github/actions/setup-action
+
+      - name: Remove files from aepsdk-commons
+        run: rm -rf ./github/aepsdk-commons
+
+      - name: Run iOS Functional Tests
+        run: make functional-test-ios
+
+  test-ios-integration:
+    runs-on: macos-latest
+    needs: validate-code
+    if: inputs.run_test_ios_integration && (inputs.github_ref == 'refs/heads/main' || inputs.github_ref == 'refs/heads/staging')
```

Comment:
Adding these checks `&& (inputs.github_ref == 'refs/heads/main' || inputs.github_ref == 'refs/heads/staging')` makes it impossible to override the default behavior. Can we instead rely solely on a boolean value and move these checks to the caller repo?

Reply:
Similar comment for `build_xcframework_and_app`